test/test_nn.py

```diff
@@ -7743,6 +7743,47 @@ def test_smoothl1loss_negative_beta_not_supported(self):
         with self.assertRaises(RuntimeError):
             F.smooth_l1_loss(torch.randn(2, 2), torch.randn(2, 2), beta=-1.0)
 
+    def test_smooth_l1_loss_vs_huber_loss(self):
+        def _test_smooth_l1_loss_vs_huber_loss_helper(beta=1.0, require_equal=True):
+            input, target = torch.randn(2, 2), torch.randn(2, 2)
+            smooth_l1_loss = torch.nn.SmoothL1Loss(beta=beta)(input, target)
+            huber_loss = torch.nn.HuberLoss(beta=beta)(input, target)
```

Comment:
Enumerate over the reduction types to test each code path, too.

Reply:
This caught a bug calling the `_Loss` constructor incorrectly.